src/tools/pkcs11-tool.c

```diff
@@ -4847,17 +4863,15 @@ derive_ec_key(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE key, CK_MECHANISM_TYPE
 	CK_BBOOL _false = FALSE;
 	CK_OBJECT_HANDLE newkey = 0;
 	CK_ATTRIBUTE newkey_template[20] = {
-		{CKA_TOKEN, &_false, sizeof(_false)}, /* session only object */
-		{CKA_CLASS, &newkey_class, sizeof(newkey_class)},
-		{CKA_KEY_TYPE, &newkey_type, sizeof(newkey_type)},
-		{CKA_SENSITIVE, &_false, sizeof(_false)},
-		{CKA_EXTRACTABLE, &_true, sizeof(_true)},
-		{CKA_ENCRYPT, &_true, sizeof(_true)},
-		{CKA_DECRYPT, &_true, sizeof(_true)},
-		{CKA_WRAP, &_true, sizeof(_true)},
-		{CKA_UNWRAP, &_true, sizeof(_true)}
-	};
-	int n_attrs = 9;
+			{CKA_TOKEN,	    &_false,	     sizeof(_false)	   }, /* session only object */
+			{CKA_CLASS,	    &newkey_class, sizeof(newkey_class)},
+			{CKA_KEY_TYPE,    &newkey_type,  sizeof(newkey_type) },
+			{CKA_SENSITIVE,	&_false,	 sizeof(_false)      },
+			{CKA_EXTRACTABLE, &_true,	  sizeof(_true)       },
+			{CKA_WRAP,	   &_true,	   sizeof(_true)	},
+			{CKA_UNWRAP,	     &_true,	     sizeof(_true)	  }
```

Comment:
This was mentioned in the previous review and not yet addressed. Please, remove the bogus WRAP/UNWRAP attributes here.

https://github.com/OpenSC/OpenSC/pull/3090/#discussion_r1541488186